app/models/miq_action.rb

```diff
@@ -222,11 +222,13 @@ def action_audit(_action, rec, inputs)
                     :message      => "Policy #{msg}: policy: [#{inputs[:policy].description}], event: [#{inputs[:event].description}]")
   end
 
-  def action_run_ansible_playbook(action, rec, _inputs)
+  def action_run_ansible_playbook(action, rec, inputs)
     service_template = ServiceTemplate.find(action.options[:service_template_id])
-    options = { :dialog_hosts => target_hosts(action, rec),
-                :initiator    => 'control' }
-    service_template.provision_request(target_user(rec), options)
+    dialog_options = { :dialog_hosts => target_hosts(action, rec) }
+    request_options = { :extra_vars => { 'event_source' => rec.href_slug,
+                                         'event_name'   => inputs[:event].try(:name) },
```

Comment:
@lfu the event_type doesn't have much information since it comes up as Default for most of the miq_event.
<img width="1220" alt="screen shot 2017-03-24 at 11 05 14 am" src="https://cloud.githubusercontent.com/assets/6452699/24300294/5b8503c8-1082-11e7-94c8-d018b16cd0c8.png">